src/Core/Grid/Query/OrderQueryBuilder.php

```diff
@@ -231,6 +230,25 @@ private function getBaseQueryBuilder(array $filters)
         return $qb;
     }
 
+    /**
+     * @param QueryBuilder $qb
+     *
+     * @return QueryBuilder
+     */
+    private function addNewCustomerField(QueryBuilder $qb)
+    {
+        $newCustomerSubSelect = $this->connection
+            ->createQueryBuilder()
+            ->select('so.id_order')
+            ->from($this->dbPrefix . 'orders', 'so')
+            ->where('so.id_customer = o.id_customer')
+            ->andWhere('so.id_order < o.id_order')
+            ->setMaxResults(1)
+        ;
+
+        return $qb->addSelect('IF ((' . $newCustomerSubSelect->getSQL() . ') > 0, 0, 1) AS new');
```

Comment:
I don't think you need to return the query builder? Since it is passed as argument and already modified